app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SessionsController < ApplicationController
+  before_action :require_logged_out, only: [:new]
+
+  # TODO: message for require_logged_out should be suppressed
+
+  def create
+    user = User.find_by(email: params[:email])
+
+    if user && user.authenticate(params[:password])
+      flash[:notice] = "Welcome, #{user.full_name}"
+      session[:user_id] = user.id
+      redirect_to home_path
+    else
+      flash[:error] = "Invalid email or password"
+      render :new
+    end
+  end
+
+  def destroy 
+    flash[:notice] = "You are logged out" 
+    session[:user_id] = nil
```

Comment:
Even tho it does not matter here, I would set flash message as the last thing prior redirect. So I would recommend to swap lines 20 and 21. The reason is - you eventually want to do the 'job' as first thing, and then set messages depending on what happened. This is more about pattern how you write the code in controllers.